tests/ui/type-alias-impl-trait/generic_duplicate_param_use5.rs

```diff
@@ -9,12 +9,9 @@ type Two<T, U> = impl Debug;
 
 fn two<T: Debug, U: Debug>(t: T, u: U) -> Two<T, U> {
     (t, u)
-    //~^ ERROR `T` doesn't implement `Debug`
```

Comment:
Why doesn't this error anymore at all?

Reply:
We now first check that there is a type mismatch between definition sites, which sets the hidden type to `TyKind::Error`

Reply:
Oh right, per-opaque we now check that for all sites first and only then run wf